assets/components/MainMenu.svelte

```diff
@@ -184,80 +198,97 @@ function renderUnread(conversation, max = 60) {
+          name={collapsedStates[connection.connection_id] ? 'chevron-right' : 'chevron-down'}
+          on:click={e => {
+          toggleSection(connection.connection_id);}}/>
+        </div>
       </Link>
-      {#each connection.conversations.toArray() as conversation}
-        <Link href="{conversation.path}" class="{conversationClassNames(connection, conversation)}">
-          {#if conversation.frozen && !conversation.is('private')}
-            <Icon name="exclamation-triangle"/>
-          {:else}
-            <Icon name="{conversation.is('private') ? 'user' : 'user-friends'}"/>
-          {/if}
-          <span>{conversation.name}</span>
-          <span class="tooltip">{$l(conversation.frozen)}</span>
-          <b class="badge" hidden="{!conversation.unread}">{renderUnread(conversation)}</b>
-        </Link>
+      <!-- Using a conditional render based on the collapsed state for the conversations of this connection -->
+      {#if !collapsedStates[connection.connection_id]}
+          {#each connection.conversations.toArray() as conversation}
+            <Link href="{conversation.path}" class="{conversationClassNames(connection, conversation)}">
+              {#if conversation.frozen && !conversation.is('private')}
+                  <Icon name="exclamation-triangle"/>
+              {:else}
+                  <Icon name="{conversation.is('private') ? 'user' : 'user-friends'}"/>
+              {/if}
+              <span>{conversation.name}</span>
+              <span class="tooltip">{$l(conversation.frozen)}</span>
+              <b class="badge" hidden="{!conversation.unread}">{renderUnread(conversation)}</b>
+            </Link>
+          {/each}
+      {/if}
       {/each}
-    {/each}
-
-    <h3>{$user.email || $l('Account')}</h3>
-    <Link href="/chat">
-      <Icon name="bell"/>
-      <span>{$l('Notifications')}</span>
-      <b class="badge" hidden="{!$notifications.unread}">{renderUnread($notifications)}</b>
-    </Link>
-    <Link href="/search">
-      <Icon name="search"/>
-      <span>{$l('Search')}</span>
-    </Link>
-    <Link href="{addConversationLink}">
-      <Icon name="comment"/>
-      <span>{$l('Add conversation')}</span>
-    </Link>
-    <Link href="/settings/connections">
-      <Icon name="network-wired"/>
-      <span>{$l('Connections')}</span>
-    </Link>
-    <Link href="/settings/account">
-      <Icon name="user-cog"/>
-      <span>{$l('Account')}</span>
-    </Link>
-    <Link href="/settings/files">
-      <Icon name="folder-open"/>
-      <span>{$l('Files')}</span>
-    </Link>
-    {#if $user.is('admin')}
-      <Link href="/settings">
-        <Icon name="tools"/>
-        <span>{$l('Settings')}</span>
+
+    <h3 class="account-toggle-plus" on:click={() => toggleSection($user.email)}>
+      <span>{$user.email || $l('Account')}</span>
+      <Icon name={collapsedStates[$user.email] ? 'plus' : 'minus'} />
+    </h3>
+    <div class={collapsedStates[$user.email] ? 'collapsedSection' : ''}>
```

Comment:
Not a fan of camelCase in classnames. But either way, I think you can just use the "hidden" class, which already exists. I would even suggest using that on the links above...

Reply:
```suggestion
    <div class="collapsedSection" hidden={!collapsedStates[$user.email]}>
```

Reply:
Actually, that's wrong, it should be:
    <div hidden={!collapsedStates[$user.email]}>

Reply:
```suggestion
    <div class={collapsedStates[$user.email] && !filter.length ? 'collapsed-section' : ''}>
```

I'm making this suggestion, as I tried using the hidden class, but then the filter/search for things like "Files" or "Settings" wouldn't work anymore.